### ISSUE TITLE
3x difference in BIN size between export in Blender 3.1.2 & 3.3.x

### ISSUE THREAD

**Issue by rotornut44** [AsoboStudio/glTF-Blender-IO-MSFS#251]
### Current Behavior

glTF files with animations exported under Blender v3.3.x have considerably larger BIN files than those exported with Blender v 3.1.x.
I tested this under Blender 3.1.2 with exporter v1.1.6, 1.3.0, & 1.3.1 (the latter two still seem to work fine, even though they are not marked as compatible) as well as Blender 3.3.3 & 3.3.8 with exporter 1.3.1.

### Expected Behavior

With the exact same export settings and no changes otherwise, I would believe that the BIN files would be of a comparable size. However, the difference is currently about 3x that of the original, which makes me wonder why so much additional data is being written?

### Steps To Reproduce

1. Open reference (esd_windsock_debug - attached below) in Blender 3.3.x.
2. Export. BIN file size will be ~2.08 MB.
3. Open reference in Blender 3.1.2 (Using v1.1.6, 1.3.0, or 1.3.1 exporter - the latter two still seem to work fine).
4. Export. BIN file size will be ~676 KB.

### Environment

```markdown
- OS: Windows 10
- Blender: 3.3.8 (Also tested on 3.3.3)
- glTF-Blender-IO-MSFS: 1.3.1
```

### Anything else?

![blender_exp](https://github.com/AsoboStudio/glTF-Blender-IO-MSFS/assets/78777891/f34fda6c-3df7-4782-8ac7-05d4981ae2c3)
[esd_windsock_debug.zip](https://github.com/AsoboStudio/glTF-Blender-IO-MSFS/files/11951024/esd_windsock_debug.zip)

**Comment by ronh991** [comment]
The ASOBO exporter uses the core Khronos gltf coding in blender.  This new code will automatically bake ALL your animations.  This may be the reason for your larger file size.  

Uncheck the "Always Sample Animations"
and
Check "Optimize Animation Size"

This is a core Khronos code issue.

![DA40_CGDFQ_2023 07 04_20h13m35s](https://github.com/AsoboStudio/glTF-Blender-IO-MSFS/assets/19254353/8ba50af6-d2c4-4129-bd1b-4c1304572e84)

**Comment by rotornut44** [comment]
I tried this method earlier based on one of the other threads here, and the result is an even greater increase in file size.
The BIN goes from 2.08 MB to 5.4 MB.

If this is just an issue with Khronos, since its only affecting models with animations, I will just export them with 3.1.2 and keep using the latest for everything else.

**Comment by ronh991** [comment]
Reviewing your zip file, I can reproduce the file size issue.  I also see that with my suggestion that the interpolation is CUBICSPLINE for the unchecked "Always Sample Animations" which I find curious.  Did you make the animations using curves, follow path?

**Comment by rotornut44** [comment]
Nope. Not that I am aware of. I used a wind simulation on the sock mesh, which I baked to the bones.
Then after dropping the NLAs, I set the keyframe interpolation to linear.

**Comment by ronh991** [comment]
Ok I redid your windsock and set all the bones individually to linear interpolation.  I can't set default or wind0 as there is no key frames.  Coul;d you test my changes.  You will see now that the bin file is smaller.

esd_windsock_debug_allchecked - has "always sample" checked

esd_windsock_debug_nonchecked - does not
[esd_windsock_debug_ron.zip](https://github.com/AsoboStudio/glTF-Blender-IO-MSFS/files/11959859/esd_windsock_debug_ron.zip)

![DA40_CGDFQ_2023 07 05_12h0m40s](https://github.com/AsoboStudio/glTF-Blender-IO-MSFS/assets/19254353/6f71a1cd-a7e9-46fb-aa92-19b769de791e)

**Comment by ronh991** [comment]
> Not that I am aware of. I used a wind simulation on the sock mesh, which I baked to the bones. Then after dropping the NLAs, I set the keyframe interpolation to linear.

I can't be sure but the wind simulation may set the animation interpolation to curves and hence you get cubicspline as interpolation output.  The sim does not like the gltf file having cubicspline. It's unfortunate that checking "always sample" results in automatically setting interpolation to LINEAR, while unchecked results in CUBICSPLINE.

To help with default and wind0 - I suggest having key frames at start and end and just use the rest position for both. That way there are key frames to set.  I do this in my bone animations and works for me.

**Comment by rotornut44** [comment]
I went back to my original (non-debug) file and the keyframes for the bones in fact did not have linear interpolation, just the rotation did. I set them all to linear and also noticed a wrench next to them in the dopesheet that says "Enable F-Curve modifiers" which I unchecked. Also, in the Action Editor, if I hover over the NLA track name, it says "A collection of F-Curves for animation."

So, it definitely is using F-Curves when you bake an animation. I confirmed this by checking in the Graph Editor. Your debug project also has them still.

I tried an export with my original file with the linear changes as well as your export recommendations, and the file size is now 7.33 MB for my file and 1.92 MB for your debug Blender. So, mine got worse, and yours is still exporting larger than what you provided in the ZIP. This was exported under Blender 3.3.8 and exporter 1.3.1.

I have looked into how to clear the F-Curves, but have been unable to find a solution that works as described. The F-Curves are apparently already sampled to keyframes, or else I supposedly wouldn't be able to edit them...
